### ISSUE TITLE
Performance: readData utterly slow for files with many lines of data

### ISSUE THREAD

**Issue by FObersteiner** [cedadev/nappy#57]
### Description

Loading data from small files completes in a decent amount of time. With many lines of data (10k+), the process becomes a "bottleneck".

### What I Did

read 4.3k lines of data, ffi1001:
```
%timeit myfile.readData()
67.9 ms ± 7.36 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

read 86.6k lines of data, ffi1001:
```
%timeit myfile.readData()
51.5 s ± 2.54 s per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

That's nearly a minute per file! If I'd want to load many such files, I'd have to go have a lot of coffee in the meantime ☕👾

---

tracing the execution of the call to `readData`, I find
- [this while loop](https://github.com/cedadev/nappy/blob/1452cebd9548b967dd6704e7688ee2af1815e721/nappy/na_file/na_file.py#L345) calls internal read method(s) for each line of data
- for NA1001, [this method](https://github.com/cedadev/nappy/blob/1452cebd9548b967dd6704e7688ee2af1815e721/nappy/na_file/na_file_1001.py#L64) calls the parser `readItemsFromUnknownLines`
- the parser has multiple conditionals, while loops etc. ...
- it [checks for curly braces](https://github.com/cedadev/nappy/blob/1452cebd9548b967dd6704e7688ee2af1815e721/nappy/utils/right_strip.py#L25) *in each line of data* with a regex ...

**Comment by agstephens** [comment]
@FObersteiner, I agree that we should look at this. Do you have publicly downloadable large example files that we could use in unit/integration testing?
